src/operations.cc

```diff
@@ -309,6 +309,20 @@ namespace sharp {
           0.0, 0.0, 0.0, 1.0));
   }
 
+  VImage Modulate(VImage image, double const brightness, double const saturation, double const hue) {
+    image = image.colourspace(VIPS_INTERPRETATION_LCH);
+
+    if (HasAlpha(image)) {
+      // Separate alpha channel
+      VImage alpha = image[image.bands() - 1];
+      return RemoveAlpha(image)
+        .linear({brightness, saturation, 1}, {0, 0, hue})
```

Comment:
I've just realised the hue addition could result in angle value greater than 360 but I think we're OK as libvips will always convert away from LCH via LAB using sin/cos so this shouldn't be a problem, phew.

Reply:
Testing with values like 350 shows that indeed, libvips handles overflow well. Otherwise I would have no idea how to do that manually.